### DISCUSSION TITLE
How to pass data to modalsView ?

### DISCUSSION THREAD

**Discussion by AEK-BKF** [rappasoft/laravel-livewire-tables#324] [Q&A]
Hello,
Great work man, thanks a lot,
Just updated the package and tried to use the function :

   ```
 public function modalsView(): string
    {
        return 'table.includes.modals';
    }
```
I'm wondering how can we pass data to a specific modal from this function ?

Thanks

**Comment by rappasoft** [comment]
You should be able to access any data that you have publically available from your component as it just loads the HTML into the DOM right after the table.

If you had:

```php
public bool $modalOpen = false;
```

Then in the modal:

```html
<div
    x-data="{
        show: @entangle('modalOpen')
   }"
>
    <div x-show="show">Modal content</div>
</div>
```

Or something of the sorts.

**Reply by AEK-BKF** [reply]
Thanks for reply,
But what I mean, how can I do something like this :
Let's say I have a component `manage` as a modal

`<livewire:master.posts.manage :post="$post" :do="$mode" />`

**Reply by rappasoft** [reply]
Hmm might be out of the scope of my simple solution, the only thing I can think of is to have 2 class properties to hold the current items you need. So on the button click that fires the modal set $currentPost and $currentMode or something and then those should be available in the view for you to pass to your livewire component. Unless I'm still not understanding.

**Reply by AEK-BKF** [reply]
Thanks for your effort to help me,
In fact this is what I'm doing now, but in this new feature  `modalsView()`, it's useful for dedicated modals for the index component only, not for reusable modals.
It will be helpful if we have a place like row view where we can put any modal from anywhere.
Thanks

**Reply by rappasoft** [reply]
Hmm I'm willing to build upon it if I understand the need. Can you give me a step by step of what you are looking to do? I'll see if it's something everyone can use.
